tgui/packages/tgui/interfaces/SeedExtractor.js

```diff
@@ -0,0 +1,94 @@
+import { useBackend } from '../backend';
+import { Button, Section, Table, Box, NumberInput } from '../components';
+import { Window } from '../layouts';
+
+export const SeedExtractor = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    stored_seeds,
+    vend_amount,
+  } = data;
+
+  return (
+    <Window resizable>
+      <Window.Content scrollable className="Layout__content--flexColumn">
+        <Section title="Stored Seeds">
+          <div className="CameraConsole__toolbarRight">
+            Set Amount to be Vended:&nbsp;
+            <NumberInput
+                animated
+                value={vend_amount}
+                width="40px"
+                minValue={1}
+                maxValue={25}
+                stepPixelSize={3}
+                onDrag={(e, value) =>
+                  act('set_vend_amount', {
+                    vend_amount: value,
+                  })
+                }
+              />
+          </div>
+          {stored_seeds?.length
+          ? <SeedsContent />
+          : "No Seeds"
+          }
+        </Section>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const SeedsContent = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    stored_seeds,
+  } = data;
+
+  return (
+    <Table>
+      <Table.Row bold>
+        <Table.Cell>Name</Table.Cell>
+        <Table.Cell>Variant</Table.Cell>
+        <Table.Cell>Lifespan</Table.Cell>
+        <Table.Cell>Endurance</Table.Cell>
+        <Table.Cell>Maturation</Table.Cell>
+        <Table.Cell>Production</Table.Cell>
+        <Table.Cell>Yield</Table.Cell>
+        <Table.Cell>Potency</Table.Cell>
+        <Table.Cell textAlign="middle">Stock</Table.Cell>
+      </Table.Row>
+      {stored_seeds.map((seed, index)=> (
+        <Table.Row key={index}>
+          <Table.Cell>
+            <img
+              src={`data:image/jpeg;base64,${seed.image}`}
+              style={{
+                'vertical-align': 'middle',
+                width: '32px',
+                margin: '0px',
+                'margin-left': '0px',
+              }}
+            />
+            {seed.name}
+          </Table.Cell>
+          <Table.Cell>{seed.variant}</Table.Cell>
```

Comment:
You know what..
Variant is a user-provided string that 1) is empty most of the time and 2) can in theory get rather long.
It's probably much better to put it into the Name column on a new line. Something like
```js
{seed.name}
{seed.variant ? (<br/> <i>{seed.variant}</i>) : ""}
```

(i'm not super sure how idiomatic the `<i>` is, but the only two alternatives i see are `<span style="font-style: italic">` and `<Box italic>`; the former looks ugly in code, and the latter might have weird padding to it. But it might also not!..)

wdyt?

Reply:
personally, if I were to do something like this I'd opt just to replace the seed name entirely with the variant, as the botanist would likely know what seed it was anyway (not that it matters since they're likely only using it for its genes). This could be colored text aswell to indicate that its not the standard name aswell. We could do this:
```js
{seed.variant ? seed.variant : <span color="green">seed.name</span>}
```
also since you already wrapped that statement in curly brackets, you don't need to further wrap `seed.variant` because react already knows you're using an expression/accessing variables.

Reply:
> also since you already wrapped that statement in curly brackets, you don't need to further wrap seed.variant because react already knows you're using an expression/accessing variables. 

That would be true if there wasn't an extra `<span ` html element there there. Since it's wrapped in curly braces that gets interpreted as "Less than `span` ..", which is invalid syntax at the very least. That's why i wrap that in brackets `()` first to switch the mode to "this is html actually", and then inside that switch it back to curly `{}`.

> replace the seed name entirely with the variant, as the botanist would likely know what seed it was anyway 

I don't think the variant names are often descriptive enough without knowing the species names. They are often set as a "modifier" of sort, like "teslium", not a full name like "teslium apples". If we don't display the base plant name, that can get mildly confusing.

> not a massive fan of line breaking here because that would look ugly on a table imo

Moving the variant to the base name in brackets (i.e. `Apple tree (teslium)`) and optionally coloring the bracket contents is fine by me too. I think it would still be better than a dedicated column.

Reply:
probably could do that, I'll see what I can make out of it. I'm busy with some other stuff atm so it may take me a day or two to get screenshots/a working version of this out